backend/src/test/java/gov/cdc/usds/simplereport/db/repository/DataHubUploadEntityRespositoryTest.java

```diff
@@ -0,0 +1,79 @@
+package gov.cdc.usds.simplereport.db.repository;
+
+import gov.cdc.usds.simplereport.config.simplereport.DataHubConfig;
+import gov.cdc.usds.simplereport.db.model.DataHubUpload;
+import gov.cdc.usds.simplereport.db.model.auxiliary.DataHubUploadStatus;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SuppressWarnings("checkstyle:MagicNumber")
+class DataHubUploadEntityRespositoryTest extends BaseRepositoryTest {
+
+    @Autowired
+    private DataHubUploadRespository _repoDH;
+    @Autowired
+    private DataHubConfig _config;
+
+    @Test
+    void testBasicSavingQueryDataHubUpload() {
+        final Date DATE_OLDEST = Date.from(Instant.parse("2000-01-01T00:00:00Z"));
+        final Date DATE_3MIN_AGO = new Date(System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(3));
+        final Date DATE_4MIN_AGO = new Date(System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(4));
+        final Date DATE_5MIN_AGO = new Date(System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(5));
+        final String TEST_RESPONSE = "{\"UnitTestResponse\":\"OK\"}";
+
+        // add a new entry, verify count increases by 1
+        List<DataHubUpload> results_before = _repoDH.findAll();
+
+        DataHubUpload d1 = new DataHubUpload(_config)
+                .setEarliestRecordedTimestamp(DATE_OLDEST)
+                .setLatestRecordedTimestamp(DATE_5MIN_AGO);
+        _repoDH.save(d1);
+
+        assertNotNull(d1.getInternalId());
+        assertEquals(d1.getCreatedAt(), d1.getUpdatedAt());
+        _repoDH.save(d1);
+
+        // Spring's updateAt doesn't work as expected.
```

Comment:
Hibernate doesn't do an update if you didn't modify the object. 🙂 

Note also that you may need to do a flush (grep for that keyword in this directory) in order to force data to be committed to the database, which can be part of the issue in situations like this.